homeassistant/components/shelly/utils.py

```diff
@@ -346,3 +346,30 @@ def get_rpc_input_triggers(device: RpcDevice) -> list[tuple[str, str]]:
             triggers.append((trigger_type, subtype))
 
     return triggers
+
+
+async def async_device_update_info(
+    hass: HomeAssistant, shellydevice: BlockDevice | RpcDevice, shellyentry: ConfigEntry
+) -> None:
+    """Update device registry info."""
+
+    _LOGGER.debug("Updating device registry info")
+    dev_registry = device_registry.async_get(hass)
+    device = None
+    if shellyentry.unique_id:
```

Comment:
Is there any way to reach here without unique id?

Reply:
Nope, but we need to make mypy happy

Reply:
We should not change code logic just to make mypy happy. mypy doesn't know we can't reach here without unique id. It would be better to have an assert here so that we don't just silently drop the error.

Reply:
Done